src/main/java/org/apache/commons/lang3/ClassUtils.java

```diff
@@ -375,18 +379,22 @@ public static String getPackageName(final Class<?> cls) {
      * @param className  the className to get the package name for, may be {@code null}
      * @return the package name or an empty string
      */
+    @SuppressWarnings("index:argument.type.incompatible") /*
+    #1, #2: !StringUtils.isEmpty(classname) => @MinLen(1), also according to the documentation, the name cannot end with '[', hence the subtring also has @MinLen(1)
```

Comment:
I think I suggested this on your other pull request, but you should put an `@EnsuresMinLenIf` annotation on `StringUtils.isEmpty` to avoid these false positives.

Reply:
I have already annotated StringUtils.isEmpty that way.

Reply:
Then why is this warning suppression needed?

Reply:
I think it is issuing a warning in #1 because of #2, after the substring, the checker assumes className may be a null string .
```
while (className.charAt(0) == '[') { // #1
            className = className.substring(1); // #2
        }
```

Reply:
That doesn't make sense. The Index Checker has no concept of nullness, so it wouldn't assume that the string is null.

It (rightly) thinks that `className` could be empty when entering the loop. How do you know that isn't the case? That would be the part of your comment about "className cannot end with a '['" - which I think should be the whole justification.